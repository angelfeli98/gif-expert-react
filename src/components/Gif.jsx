
import Proptypes from 'prop-types';

function Gif({gif}) {
  return (
    <li className="card">
        <img
            src={gif.url}
            alt={gif.title}
        />
        <p> { gif.title } </p>
    </li>
  )
}

Gif.propTypes = {
    gif: Proptypes.object.isRequired,
}

export default Gif;