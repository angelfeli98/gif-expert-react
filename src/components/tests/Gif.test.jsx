/* eslint-disable no-undef */

import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import Gif from '../Gif';

let gif = {
    title: 'Mira este gif',
    url: 'https://url.example.com',
}

describe('<Gif />' , () => {
    test('should render', () => {
        const {container} = render(<Gif gif={gif} />);
        expect(container).toMatchSnapshot();
    })

    test('should render props porpertly', () => {
        const component = render(<Gif gif={gif} />);
        expect(component.getByRole('img')).toHaveAttribute('src', gif.url);
        expect(component.getByRole('img')).toHaveAttribute('alt', gif.title);
        expect(component.getByText(gif.title).closest('p')).toBeInTheDocument();
    })
})