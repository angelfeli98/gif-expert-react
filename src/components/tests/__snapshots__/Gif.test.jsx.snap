// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Gif /> should render 1`] = `
<div>
  <li
    class="card"
  >
    <img
      alt="Mira este gif"
      src="https://url.example.com"
    />
    <p>
       
      Mira este gif
       
    </p>
  </li>
</div>
`;
